src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -269,6 +294,68 @@ public DetailAST getUnderlyingNode() {
         return detailAst;
     }
 
+    /**
+     * Returns preceding sibling axis iterator.
+     * @return iterator
+     */
+    private AxisIterator getPrecedingSiblingsIterator() {
+        final AxisIterator result;
+        if (positionAmongSiblings == 0) {
+            result = EmptyIterator.OfNodes.THE_INSTANCE;
+        }
+        else {
+            try (AxisIterator iterator = new ArrayIterator.OfNodes(
+                    getPrecedingSiblings().toArray(EMPTY_ABSTRACT_NODE_ARRAY))) {
+                result = iterator;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Returns following sibling axis iterator.
+     * @return iterator
+     */
+    private AxisIterator getFollowingSiblingsIterator() {
+        final AxisIterator result;
+        if (positionAmongSiblings == parent.getChildren().size() - 1) {
+            result = EmptyIterator.OfNodes.THE_INSTANCE;
+        }
+        else {
+            try (AxisIterator iterator = new ArrayIterator.OfNodes(
+                    getFollowingSiblings().toArray(EMPTY_ABSTRACT_NODE_ARRAY))) {
+                result = iterator;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Returns following siblings of the current node.
+     * @return siblings
+     */
+    private List<AbstractNode> getFollowingSiblings() {
+        final List<AbstractNode> siblings = new ArrayList<>();
+        for (int i = positionAmongSiblings + 1; i < parent.getChildren().size(); i++) {
```

Comment:
Is it the same as
```
final List<AbstractNode> siblings = parent.getChildren();
return siblings.subList(positionAmongSiblings, siblings.size());
```
? The list returned by parent.getChildren() is immutable, so it is safe to use sublist here.

Reply:
done

Reply:
verified, done